### PR TITLE
Add a component to display the list of models from gltf-Sample-Models and let you load them

### PR DESCRIPTION
**Pull request by AdamMitchell-ms**
- Add a component to download and parse model-index.json from:
  - https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/model-index.json
- A custom inspector from that component will display the list of models in the Unity Editor and allow you to load models onClick.

### CODE REVIEW

File: UnityGLTF/Assets/UnityGLTF/Scripts/Tests/Editor/SampleModelListInspector.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		var manifestRelativePath = serializedObject.FindProperty(SampleModelList.ManifestRelativePathFieldName).stringValue;
+
+		var loader = new WebRequestLoader(pathRoot);
+		await loader.LoadStream(manifestRelativePath);
+
+		loader.LoadedStream.Seek(0, SeekOrigin.Begin);
+
+		var streamReader = new StreamReader(loader.LoadedStream);
+
+		//var s = streamReader.ReadToEnd();
```
Comment (admachina) [inline UnityGLTF/Assets/UnityGLTF/Scripts/Tests/Editor/SampleModelListInspector.cs]: Nit: Remove

File: UnityGLTF/Assets/UnityGLTF/Scripts/Tests/SampleModelList.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	public static string PathRootFieldName => nameof(pathRoot);
+	public static string ManifestRelativePathFieldName => nameof(manifestRelativePath);
+	public static string ModelRelativePathFieldName => nameof(modelRelativePath);
+	public static string LoadThisFrameFieldName => nameof(loadThisFrame);
+
+	[SerializeField]
+	private GLTFComponent loader = null;
+
+	[SerializeField]
+	private string pathRoot = "http://localhost:8080/glTF-Sample-Models/2.0/";
```
Comment (pjramirez-ms) [inline UnityGLTF/Assets/UnityGLTF/Scripts/Tests/SampleModelList.cs]: Code still works if this folder doesn't exist?

File: UnityGLTF/Assets/UnityGLTF/Scripts/Tests/Editor/SampleModelListInspector.cs (hunk 1 above)
Comment (AdamMitchell-ms) [reply UnityGLTF/Assets/UnityGLTF/Scripts/Tests/Editor/SampleModelListInspector.cs]: Good catch.

File: UnityGLTF/Assets/UnityGLTF/Scripts/Tests/SampleModelList.cs (hunk 2 above)
Comment (AdamMitchell-ms) [reply UnityGLTF/Assets/UnityGLTF/Scripts/Tests/SampleModelList.cs]: It gives a generic error.  I've added a nice error that describes what went wrong.  See my changes to SampleModelListInspector.cs

**Pull request merged**